### ISSUE TITLE
[Task] CLI SAMM2AASX Add qualifier for cardinality

### ISSUE THREAD

**Issue by BirgitBoss** [eclipse-esmf/esmf-sdk#435]
**Is your task related to a problem? Please describe.**

Up to now no qualifiers are added because SAMM defines these qualifiers like cardinality etc.
See https://industrialdigitaltwin.org/wp-content/uploads/2022/12/I40-IDTA-WS-Process-How-to-write-a-SMT-FINAL-.pdf Chapter 7.2

Nevertheless it might be useful to generate those as well to be consistent with the published other submodel templates.

**Describe the solution you'd like**
For all optional elements add qualifier **Cardinality** with value **ZeroToOne**:

```
          <qualifiers>
            <qualifier>
              <semanticId>
                <type>ExternalReference</type>
                <keys>
                  <key>
                    <type>GlobalReference</type>
                    <value>https://admin-shell.io/SubmodelTemplates/Cardinality/1/0</value>
                  </key>
                </keys>
              </semanticId>
              <type>Cardinality</type>
              <valueType>xs:string</valueType>
              <value>ZeroToOne</value>
            </qualifier>
          </qualifiers>
```

**Comment by BirgitBoss** [comment]
Hint, might help: see https://github.com/admin-shell-io/aas-specs/tree/master/schemas/xml/examples/generated for generated valid xml files.

**Comment by KobOp** [comment]
Dear @BirgitBoss,

IMHO, the condition for "optional" elements can also be ZeroOrMany (0 or 1+), in addition to ZeroToOne, unless the maximum value is specified to 1 (or the value is not a list).

**Comment by BirgitBoss** [comment]
Dear @KobOp , sets and lists etc. are mapped to SubmodelElementList (SML) and SML implicitly have cardinality 0..*. This is why I feel we only need cardinality qualifiers for optional SAMM properties.

**Comment by chris-volk** [comment]
Dear @BirgitBoss , since optionality in SAMM is not attributed to a Property itself, but in its usage by an Entity or the Aspect itself, a Property can be used mandatory in one context and optionally in another context. This "property usage" does not exist in AAS.
Therefore, the choices we propose are as follows:
1. Keep current documentation (ZeroToOne is not included, even if a property is used optionally)
2. Add the ZeroToOneQualifier whenever there is an optional usage of a Property (even though it may also be mandatory in another usage which may be in the same aspect model that you are currently converting)
3. Create disambiguating clones of Properties (complex, one clone per usage? one clone for mandatory usage + one for optional usage? Probably best to only create clones if mandatory+optional occur, and then create only two clones, one for optional+one for mandatory)

**Comment by chris-volk** [comment]
Where is ZeroToOne being used in the further AAS tooling, do you have any insights on that?

**Comment by BirgitBoss** [comment]
The AAS property is not the AAS concept description. They are completely separate. Therefore the concept description represents the  SAMM property and the AAS property (that is not reusable) represents the context.

**Comment by atextor** [comment]
@BirgitBoss Yes, but the problem still exists: If I understand the AAS meta model correctly, Qualifiers are attached to submodel elements, not to concept descriptions. In SAMM, the _usage_ of a Property can be optional, not the Property itself. Let's assume we have an Aspect Model with a Property _p_. In SAMM->AASX, a corresponding AAS property _p'_ is created, with a concept description that represents _p_. Now _p_ is used twice in the Aspect Model, let's say it's mandatory in the Aspect itself, and optional in one of the Entities that are used in the Aspect. In the AASX, _p'_ is still only created once, and still only one concept description is created. But we can't add a ZeroToOne Qualifier to _p'_, because it's not always optional. Even if we would create two concept descriptions for _p'_, we could not attach the Qualifier to one of them, because Qualifiers don't belong on concept descriptions.
